message_definitions/v1.0/common.xml

```diff
@@ -4935,6 +4935,24 @@
       <extensions/>
       <field type="float[58]" name="data">data</field>
     </message>
+    <!-- Smart battery messages -->
+    <message id="370" name="SMART_BATTERY_INFO">
+      <wip/>
+      <!-- This message is work-in-progress and it can therefore change. It should NOT be used in stable production environments. -->
+      <description>Smart Battery information (static/infrequent update). Use for updates from: smart battery to flight stack, flight stack to GCS. Use BATTERY_STATUS for smart battery frequent updates.</description>
+      <field type="uint8_t" name="id" instance="true">Battery ID</field>
+      <field type="uint8_t" name="battery_function" enum="MAV_BATTERY_FUNCTION">Function of the battery</field>
+      <field type="uint8_t" name="type" enum="MAV_BATTERY_TYPE">Type (chemistry) of the battery</field>
+      <field type="int32_t" name="capacity_full_specification" units="mAh">Capacity when full according to manufacturer, -1: field not provided.</field>
+      <field type="int32_t" name="capacity_full" units="mAh">Capacity when full (accounting for battery degradation), -1: field not provided.</field>
+      <field type="uint16_t" name="cycle_count">Charge/discharge cycle count. UINT16_MAX: field not provided.</field>
+      <field type="char[16]" name="serial_number">Serial number in ASCII characters, 0 terminated. All 0: field not provided.</field>
+      <field type="char[50]" name="device_name">Static device name. Encode as manufacturer and product names separated using an underscore.</field>
+      <field type="uint16_t" name="weight" units="g">Battery weight. 0: field not provided.</field>
+      <field type="uint16_t" name="discharge_minimum_voltage" units="mV">Minimum per-cell voltage when discharging. If not supplied set to UINT16_MAX value.</field>
```

Comment:
if this is per-cell, then where do we say the number of cells? I think we should have a uint8_t num_cells

Reply:
Makes sense to add that. The other item I thought about is discharge_maximum_current?

Reply:
Remember that this message doesn't replace     <message id="147" name="BATTERY_STATUS"> but supplements it, so I think number of cells should be pulled out of the cell voltages sent in that message.  Discharge_maximum_current is potentially problematic, as it varies with voltage, and smart batteries should be managing this anyway.  I agree it would be useful in some circumstances though.

Reply:
Discharge_current_maximum doesn't vary with voltage as it is a manufacturer specification.  Perhaps you meant discharge_maximum_power? Only reason I bring it up is that this message and  message 147 seems to capture most of the important info. regarding a pack except the max. discharge current. And I saw the use of a few parameters to limit motor power based upon maximum discharge.

Reply:
Notably nothing in the message seems to tell you what maximum voltage is...

Reply:
I think the intent was that max voltage is inferred from chemistry and number of cells, but valid point.
We’re starting to scope creep though.

Reply:
I'm wondering if the charging_minimum_voltage was really meant to be charging_maximum_voltage? It is a little odd for an autopilot to care about that as usually is it only for chargers to prevent charging of a deeply discharged battery.

Reply:
@hendjoshsr71 on the min-voltage thing - hybrid vehicles are a thing...

Reply:
Added these items